examples/adc-dma-rx.rs

```diff
@@ -34,21 +47,51 @@ fn main() -> ! {
 
     // Configure pa0 as an analog input
     let adc_ch0 = gpioa.pa0.into_analog(&mut gpioa.crl);
+    let adc_ch2 = gpioa.pa2.into_analog(&mut gpioa.crl);
+
+    // single-channel continuous conversion
+    let (adc1, adc_ch0, dma_ch1) = {
+        let adc_dma = adc1.with_dma(adc_ch0, dma_ch1);
+        let buf = singleton!(: [u16; 8] = [0; 8]).unwrap();
+
+        // The read method consumes the buf and self, starts the adc and dma transfer and returns a
+        // RxDma struct. The wait method consumes the RxDma struct, waits for the whole transfer to
+        // be completed and then returns the updated buf and underlying adc_dma struct. For non
+        // blocking, one can call the is_done method of RxDma and only call wait after that method
+        // returns true.
+        let (_buf, adc_dma) = adc_dma.read(buf).wait();
+        hprintln!("single-channel continuous conversion={:?}", _buf).ok();
+
+        // Consumes the AdcDma struct, restores adc configuration to previous state and returns the
+        // Adc struct in normal mode.
+        adc_dma.split()
+    };
+
+    // multi-channel single-shot conversion
+    let (adc1, AdcPins(adc_ch0, adc_ch2), dma_ch1) = {
+        let pins = AdcPins(adc_ch0, adc_ch2);
+        let adc_dma = adc1.with_scan_dma::<_, adc::Single>(pins, dma_ch1);
+
+        // TODO: should match with # of conversions specified by AdcPins
```

Comment:
I tried to add validation for ADC DMA buffer length
 - When a buffer is larger than ADC conversion length, DMA will fill the buffer partially.
 - When a buffer is smaller than ADC conversion length, it will panic. I'm not sure if it's a best way to handle the case.

Reply:
Hmm, I'm not sure what the best behaviour would be here. I'd almost argue that a `panic` is best in both cases (unless we can return a result).